bin/public/installLdapmysql.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+'''
+Ldap server with mysql backend
+
+'''
+
+__author__ = "Mattias.hemmingsson@fareoffice.com, daniel@cybercow.se"
+__copyright__ = "Copyright 2012, The System Console project"
+__maintainer__ = "Daniel Lindh"
+__email__ = "syco@cybercow.se"
+__credits__ = ["???"]
+__license__ = "???"
+__version__ = "1.0.0"
+__status__ = "Production"
+
+import os
+
+from general import x,download_file,generate_password
+import scopen
+import app
+import config
+import version
+import string
+
+
+# The version of this module, used to prevent the same script version to be
+# executed more then once on the same host.
+SCRIPT_VERSION = 1
+
+def build_commands(commands):
+    commands.add("install-ldapmysql", install_ldapmysql, help="Install Openldap server with mysql backend")
+    commands.add("unistall-ldapmysql", unistall_ldapmysql, help="Uninstall Openldap Server with mysql backend")
+
+
+def install_ldapmysql(args):
+    '''
+    Install and configure openldap server with mysql backend
+
+    '''
+    if os.path.exists("/etc/openldap/slapd.conf"):
+
+      print("There is a slapd-config file will not install")
+
+    else:
+      '''
+      No configfile move on
+      '''
+      password =  generate_password(length=16,chars=string.letters + string.digits)
+      x('yum install wget make gcc  mysql-devel unixODBC-devel groff -y')
+      os.chdir('/opt')
+      x('wget ftp://mirror.switch.ch/mirror/OpenLDAP/openldap-release/openldap-2.4.43.tgz')
```

Comment:
you should use the install.download_file (might have another name, you find it in other install scripts)... 
Then the script will download the tgz script into a temprary folder that are automatically deleteted.